libevmasm/Assembly.cpp

```diff
@@ -1636,6 +1646,16 @@ LinkerObject const& Assembly::assembleEOF() const
 			case RetF:
 				ret.bytecode.push_back(static_cast<uint8_t>(Instruction::RETF));
 				break;
+			case SwapN:
+				ret.bytecode.push_back(static_cast<uint8_t>(Instruction::SWAPN));
+				solAssert(item.data() >= 1 && item.data() <= 256);
+				ret.bytecode.push_back(static_cast<uint8_t>(item.data() - 1));
+				break;
+			case DupN:
+				ret.bytecode.push_back(static_cast<uint8_t>(Instruction::DUPN));
+				solAssert(item.data() >= 1 && item.data() <= 256);
+				ret.bytecode.push_back(static_cast<uint8_t>(item.data() - 1));
+				break;
```

Comment:
You can combine these:
```suggestion
			case SwapN:
			case DupN:
				ret.bytecode.push_back(static_cast<uint8_t>(item.instruction()));
				solAssert(item.data() >= 1 && item.data() <= 256);
				ret.bytecode.push_back(static_cast<uint8_t>(item.data() - 1));
				break;
```

Reply:
We should also assert under `case Operation` against these instructions. Nothing currently prevents you from creating an `Operation` item with one of the new instructions if you don't know you should use an assembly item and such an operation would probably be just silently accepted and just handled incorrectly.

BTW such sanity check would really be better off in `AssemblyItem` constructor but that class has a ton of constructors and the instruction is also sometimes set after construction anyway.

Reply:
Yeah, we should also have an instruction-level trait for instructions that should only occur as operation in ``AssemblyItem`` and assert during constructions that they don't occur otherwise or something like that - but also separate thing.